### ISSUE TITLE
nix: Provide & use overlay for packages?

### ISSUE THREAD

**Issue by antifuchs** [eikek/docspell#612]
I'm really grateful that you're providing nix modules for docspell - it's much easier to configure than paperless/-ng that way (-:

As I'm currently having trouble installing docspell in a container due to confusion around the `packageOverrides` solution for getting the package definitions onto nixpkgs, I feel it would be really convenient if you could provide an overlay.nix file (or attribute from release.nix) that provides a package [overlay](https://nixos.wiki/wiki/Overlays) instead of having all users install that packageOverrides configuration in their system config.

Would that be interesting to you? As I find myself with a bit of spare time, I could try to patch that in.

**Comment by eikek** [comment]
Oh definitely! The only reason for the current state is that I don't know it any better. I have to read first how overlays work. I know they are around for quite some time, but somehow my ignorance was stronger ;-) So if you find this spare time, that would be great! 
A quick question from a (very!) quick look (thanks for the link): the current package definitions can be reused in an overlay.nix, right? Is it possible to contribute the modules too that way, afaics this affects nixpkgs only?

**Comment by antifuchs** [comment]
> the current package definitions can be reused in an overlay.nix, right?

That's correct - I was going to suggest that I'd restructure pkg.nix a little, so that the function takes a store path as input; that way, we can have the default refer to the source dir that we're looking at.

And to your other question, this won't affect the nixos modules at all, correct (:
